src/components/DateInput/dateInput.test.js

```diff
@@ -54,88 +56,84 @@ describe('DateInput component', () => {
     });
 
-    it('does not call onDateChange when user types an INCOMPLETE freeform date', () => {
+    it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/3' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/3')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
-    it('does not call onDateChange when user types an INVALID freeform date', () => {
+    it('does not call onDateChange when user types an INVALID freeform date', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '2/30/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
       expect(dateChangeFn).not.toHaveBeenCalled();
     });
 
     it('calls onDateChange when user selects a day in the picker', () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      wrapper.find('Day').at(7).simulate('click');
+      fireEvent.click(screen.getByText('7'));
       expect(dateChangeFn).toHaveBeenCalled();
     });
 
     it('updates inupt value when user selects a day in the picker', () => {
-      wrapper.find('Day').at(7).simulate('click');
-      expect(input.instance().value).toHaveLength(10); // MM/DD/YYYY === 10 chars
+      fireEvent.click(screen.getByText('7'));
+      expect(screen.getByRole('textbox').value).toBe('07/07/2020'); // MM/DD/YYYY === 10 chars
+      expect(screen.getByRole('textbox').value).toHaveLength(10); // MM/DD/YYYY === 10 chars
     });
 
-    it('sets DayPicker selected date correctly from input', () => {
-      input.simulate('change', { target: { value: '4/20/2020' } });
-      const dayPickerDate = wrapper.find('DayPicker').prop('month');
-      expect(dayPickerDate.getMonth()).toBe(3);
-      expect(dayPickerDate.getFullYear()).toBe(2020);
+    it('sets DayPicker selected date correctly from input', async () => {
+      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
+      expect(screen.getByRole('gridcell', { selected: true, name: 'Mon Apr 20 2020' })).toBeTruthy()
     });
 
-    it('clears selected date when user backspaces out the input', () => {
-      const dayPickerSelectedDays = wrapper.find('DayPicker').prop('selectedDays');
+    it('clears selected date when user backspaces out the input', async () => {
       expect(dateChangeFn).not.toHaveBeenCalled();
-      input.simulate('change', { target: { value: '4/20/2020' } });
+      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
       expect(dateChangeFn).toHaveBeenCalled();
-      input.simulate('change', { target: { value: '' } });
+
+      expect(screen.getByRole('gridcell', { selected: true, name: 'Mon Apr 20 2020' })).toBeTruthy()
+
+      await userEvent.type(screen.getByRole('textbox'), '{selectall}')
+      await userEvent.type(screen.getByRole('textbox'), '{backspace}')
```

Comment:
https://github.com/testing-library/user-event

This is a separate library that enables a lot of more advanced user interactions (file uploads, selecting / deselecting, hovering, pasting, tabbing, etc.

There's a single command to clear the text as well but I like the clarity of doing it this way.

Reply:
this is great